File: android/app/src/main/java/com/wellcollective/app/WellCheckWidgetPlugin.java
package com.wellcollective.app;

import android.content.Context;
import android.content.SharedPreferences;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "WellCheckWidget")
public class WellCheckWidgetPlugin extends Plugin {
  static final String PREFS = "well_check_widget";

  @PluginMethod
  public void saveSnapshot(PluginCall call) {
    Context context = getContext();
    SharedPreferences.Editor editor = context.getSharedPreferences(PREFS, Context.MODE_PRIVATE).edit();

    editor.putString("points", call.getString("points", "--"));
    editor.putString("areas", call.getString("areas", "--"));
    editor.putString("sleep", call.getString("sleep", "--"));
    editor.putString("energyIn", call.getString("energyIn", "--"));
    editor.putString("energyOut", call.getString("energyOut", "--"));
    editor.putString("steps", call.getString("steps", "--"));
    editor.putString("updatedAt", call.getString("updatedAt", ""));
    editor.apply();

    WellCheckWidgetProvider.refreshWidgets(context);

    JSObject result = new JSObject();
    result.put("ok", true);
    call.resolve(result);
  }
}

File: ios/App/App/WellCheckWidgetPlugin.swift
import Capacitor
import Foundation
import WidgetKit

private let wellCheckWidgetAppGroup = "group.com.wellcollective.app"

@objc(WellCheckWidgetPlugin)
public class WellCheckWidgetPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "WellCheckWidgetPlugin"
    public let jsName = "WellCheckWidget"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "saveSnapshot", returnType: CAPPluginReturnPromise)
    ]

    @objc func saveSnapshot(_ call: CAPPluginCall) {
        guard let defaults = UserDefaults(suiteName: wellCheckWidgetAppGroup) else {
            call.reject("WELL Check widget storage is unavailable.")
            return
        }

        [
            "points",
            "areas",
            "sleep",
            "energyIn",
            "energyOut",
            "steps",
            "updatedAt",
            "reminder",
            "unreadCount"
        ].forEach { key in
            if let value = call.getString(key) {
                defaults.set(value, forKey: key)
            }
        }

        defaults.synchronize()
        WidgetCenter.shared.reloadAllTimelines()
        call.resolve(["ok": true])
    }
}
